### ISSUE TITLE
Fatal Crash in DownloadManager.java

### ISSUE THREAD

**Issue by logituitrel** [google/ExoPlayer#8419]
We are facing a crash captured in crashlytics.
We are not sure about the steps to replicate this.
Are there any steps to tackle these crashes/ are they fixed in any upcoming exoplayer release?
Exoplayer version: 2.11.8

Logs:
```
Fatal Exception: java.lang.IllegalStateException
at com.google.android.exoplayer2.util.Assertions.checkState(Assertions.java:81)
at com.google.android.exoplayer2.offline.DownloadManager$InternalHandler.putDownloadWithState(DownloadManager.java:1159)
at com.google.android.exoplayer2.offline.DownloadManager$InternalHandler.onRemoveTaskStopped(DownloadManager.java:1117)
at com.google.android.exoplayer2.offline.DownloadManager$InternalHandler.onTaskStopped(DownloadManager.java:1078)
at com.google.android.exoplayer2.offline.DownloadManager$InternalHandler.handleMessage(DownloadManager.java:728)
at android.os.Handler.dispatchMessage(Handler.java:110)
at android.os.Looper.loop(Looper.java:203)
at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by ojw28** [comment]
It looks like this crash can occur when you remove a download and _immediately_ re-add it (by calling `DownloadManager.addDownload` or `DownloadService.sendAddDownload`) with a _non-zero_ stop reason.

We will work on a fix. Thanks for reporting!

**Comment by ojw28** [comment]
This is fixed by the change ref'd above. It will be included in `2.12.3`. We don't have a set release date, but I would expect it to be toward the end of this month.
